### DISCUSSION TITLE
Feature request: More options in advanced password generator

### DISCUSSION THREAD

**Discussion by BrerRabbit999** [keepassxreboot/keepassxc#9658] [Ideas]
I just created a password for a site that had rather unusual criteria, and I needed to jump through hoops to trick KeepassXC into generating a valid (in their their eyes) PW.

The hardest one was to have two digits. I found a work-around, but it would be easier if there were options to include more than one character of each type.  

They also specifically required no more than one space and other odd requirements that were mostly things to avoid, so they are easier to comply.

**Comment by droidmonkey** [comment]
We won't be adjusting our generator. These are anti patterns and strongly recommended to be avoided. You'll just have to hand craft passwords for poorly implemented services if the generator doesn't nail every requirement. You can manually edit any password generated as needed.

**Reply by e-t-l** [reply]
I believe the original KeePass program had an option for a pattern-generated pw, where a pattern could be specified with regex (or a very similar-looking syntax, at least). I couldn't find that option in KeePassXC, though. Does it still exist?

**Reply by GanoItouch-Ecuador** [reply]
Hello everyone,

It would be great to improve this feature as there are devices such as routers and/or firewalls that ask you for a password format that contains a specific number of letters, numbers, uppercase, lowercase and special characters.

**Comment by SheikYahbouti** [comment]
I agree with the other posters wanting more generator options. This is a big deal when you are entering a password on your phone. If the site requires 2 lower, 2 upper, a digit and some special characters, it's easier on the phone keyboard if you can constrain the PW to something like llUUnnnn# so you aren't randomly toggling the case or long pressing every other item. Yeah they may be SLIGHTLY less secure but really, shouldn't that decision be in the hands of the actual user?

**Reply by droidmonkey** [reply]
If you want to _constrain_ a password to that degree then just type it in yourself. You don't need a generator to produce highly patterned passwords. That is the exact opposite of what a generator is doing.
